modules/transactions/src/main/java/org/apache/ignite/internal/tx/storage/state/rocksdb/TxStateRocksDbTableStorage.java

```diff
@@ -168,7 +171,7 @@ private void checkPartitionId(int partitionId) {
     }
 
     /** {@inheritDoc} */
-    @Override public CompletableFuture<Void> destroyTxStateStorage(int partitionId) throws StorageException {
+    @Override public void destroyTxStateStorage(int partitionId) throws StorageException {
```

Comment:
Here are details. This is how `destroyPartition` looks. It does have its async part.
```
    /** {@inheritDoc} */
    @Override
    public CompletableFuture<Void> destroyPartition(int partitionId) throws StorageException {
        checkPartitionId(partitionId);

        RocksDbMvPartitionStorage mvPartition = partitions.getAndSet(partitionId, null);

        if (mvPartition == null) {
            return CompletableFuture.completedFuture(null);
        }

        //TODO IGNITE-17626 Destroy indexes as well...
        mvPartition.destroy();

        // Wait for the data to actually be removed from the disk and close the storage.
        return awaitFlush(false)
                .whenComplete((v, e) -> {
                    try {
                        mvPartition.close();
                    } catch (Exception ex) {
                        LOG.error("Error when closing partition storage for partId = {}", ex, partitionId);
                    }
                });
    }
```

Reply:
Probably `close()` should be invoked synchronously, not in `whenComplete`. I'm not sure at the moment